### ISSUE TITLE
[java] TestClassWithoutTestCases reports wrong classes in a file

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3624]
**Affects PMD Version:** 6.40.0

**Rule:** [TestClassWithoutTestCases](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#testclasswithouttestcases)

**Description:**

When a test class contains additional helper classes in the same compilation units, then these are reported as well, even if they are no test classes.

**Code Sample demonstrating the issue:**

```java
import org.junit.Test;

public class MyTest {
    @Test
    public void myTest() {
    }
}

class Helper { }  // this is reported, but this is not a test class, so doesn't need test cases
```

**Expected outcome:**

PMD reports a violation at line 9, but that's wrong. That's a false positive.
